activesupport/test/cache/behaviors/local_cache_behavior.rb

```diff
@@ -17,6 +17,21 @@ def test_clear_also_clears_local_cache
     assert_nil @cache.read("foo")
   end
 
+  def test_cleanup_clears_local_cache_but_not_remote_cache
+    skip unless @cache.class.instance_methods(false).include?(:cleanup)
+
+    @cache.with_local_cache do
+      @cache.write("foo", "bar")
+      assert_equal "bar", @cache.read("foo")
+
+      @cache.send(:bypass_local_cache) { @cache.write("foo", "baz") }
+      assert_equal "bar", @cache.read("foo")
+
+      @cache.cleanup
+      assert_equal "baz", @cache.read("foo")
+    end
+  end
+
```

Comment:
The call to `cleanup` passes through to the underlying store, but the local cache is cleared.

I think it makes sense - it's probably cheaper to just empty the local cache than to iterate through all of its keys and only remove ones that have expired, on the off chance that we end up reusing the ones we keep.